Mlem/App/Views/Shared/Palette Components/PaletteForm.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  PaletteForm.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-08-30.
+//
+
+import Foundation
+import SwiftUI
+
+/// Identical to Form, but respects Palette
+struct PaletteForm<Content: View>: View {
+    @Environment(Palette.self) var palette
+    
+    @ViewBuilder let content: () -> Content
+    
+    var body: some View {
+        Form {
+            content()
+                .foregroundStyle(palette.primary)
```

Comment:
Doing this means that form headers and footers are no longer `.secondary`. Do we just accept that consequence, implement a custom `Section` too, or add `.foregroundStyle` to all headers and footers?

Reply:
Nice catch! In one iteration I actually already had a custom `Section`, I can easily dig that back up